app/src/main/java/com/rockthevote/grommet/ui/eventFlow/SessionTimeTrackingViewModel.kt

```diff
@@ -84,22 +106,57 @@ class SessionTimeTrackingViewModel(
         }
     }
 
+    fun clockIn() {
+        viewModelScope.launch(dispatchers.io + coroutineExceptionHandler) {
+            val session = sessionDao.getCurrentSession() ?: return@launch
+            val newSessionData = session.copy(clockInTime = Date())
+            sessionDao.updateSession(newSessionData)
+        }
+    }
+
+    fun clockOut() {
+        viewModelScope.launch(dispatchers.io + coroutineExceptionHandler) {
+            val session = sessionDao.getCurrentSession() ?: return@launch
+            val newSessionData = session.copy(clockOutTime = Date())
+            sessionDao.updateSession(newSessionData)
+
+            val inOutTimes = ClockInOutTimes(
+                    newSessionData.clockInTime,
+                    newSessionData.clockOutTime
+            )
+            _clockOutData.postValue(inOutTimes)
+        }
+    }
+
+    private fun updateSessionStatus() {
+        viewModelScope.launch(dispatchers.io) {
```

Comment:
We're already updating this in the `EventFlowWizard.java` class, this will trigger the event twice since we also update the status in `SessionTimeTracking`

Reply:
This doesn't update anything in shared prefs, it just retrieves the value and updates the live data object for observing the session. 

Right now, this only triggers `updateUI` which toggles the "Clock In/Out" button, which is necessary to show that the user is actually clocked in, otherwise the "Clock In" button is always displayed.

Is there another way to do this with a mechanism that's already in place?